CRM/Contribute/Form/Task/PDF.php

```diff
@@ -187,6 +195,14 @@ public function postProcess() {
       $objects['contribution']->receive_date = CRM_Utils_Date::isoToMysql($objects['contribution']->receive_date);
 
       $values = array();
+      if (isset($params['fromEmailAddress']) && !$elements['createPdf']) {
+        $fromEmail = $params['fromEmailAddress'];
+        $from = CRM_Utils_Array::value($fromEmail, $this->_emails);
+        $fromDetails = explode(' <', $from);
```

Comment:
The problem is when you load the from addresses from the optionlist it in the format of "Name" <email>" and there are array keys one for the name and one for the email in the receipt so there has to be this conversion